src/lib/utils/wrap.js

```diff
@@ -0,0 +1,11 @@
+export default function wrap(Type, fn, dontCloneWithNoArgs) {
```

Comment:
I'm curious to know if other folks think this function is sensible.  It provides a conveniently concise and reusable API, but I fear the implementation might be a little hard for us humans to parse.